google-cloud-pubsublite/src/test/java/com/google/cloud/pubsublite/internal/TopicBacklogReaderImplTest.java

```diff
@@ -0,0 +1,189 @@
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public final class TopicBacklogReaderImplTest {
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock TopicStatsClient mockClient;
+
+  private TopicPath topicPath;
+  private TopicBacklogReader reader;
+
+  @Before
+  public void setUp() throws Exception {
+    this.topicPath =
+        TopicPaths.newBuilder()
+            .setProjectNumber(ProjectNumber.of(4))
+            .setTopicName(TopicName.of("test"))
+            .setZone(CloudZone.parse("us-central1-b"))
+            .build();
+    this.reader = new TopicBacklogReaderImpl(mockClient, topicPath);
+  }
+
+  @Test
+  public void computeMessageStats_partialFailure() throws Exception {
+    ComputeMessageStatsResponse partition1 = ComputeMessageStatsResponse.getDefaultInstance();
+
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(1), Offset.of(10), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition1));
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(2), Offset.of(20), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFailedFuture(new StatusException(Status.UNAVAILABLE)));
+
+    ImmutableMap.Builder<Partition, Offset> builder = ImmutableMap.builder();
+    ApiFuture<ComputeMessageStatsResponse> future =
+        reader.computeMessageStats(
+            builder
+                .put(Partition.of(1), Offset.of(10))
+                .put(Partition.of(2), Offset.of(20))
+                .build());
+
+    ExecutionException ex = assertThrows(ExecutionException.class, future::get);
+    assertThat(ExtractStatus.extract(ex.getCause()).get().getCode()).isEqualTo(Code.UNAVAILABLE);
+  }
+
+  @Test
+  public void computeMessageStats_aggregatesEmptyMessages() throws Exception {
+    ComputeMessageStatsResponse partition1 = ComputeMessageStatsResponse.getDefaultInstance();
+    ComputeMessageStatsResponse partition2 = ComputeMessageStatsResponse.getDefaultInstance();
+    ComputeMessageStatsResponse aggregate = ComputeMessageStatsResponse.getDefaultInstance();
+
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(1), Offset.of(10), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition1));
+    when(mockClient.computeMessageStats(
+            topicPath, Partition.of(2), Offset.of(20), Offset.of(Integer.MAX_VALUE)))
+        .thenReturn(ApiFutures.immediateFuture(partition2));
+
+    ImmutableMap.Builder<Partition, Offset> builder = ImmutableMap.builder();
+    ApiFuture<ComputeMessageStatsResponse> future =
+        reader.computeMessageStats(
+            builder
+                .put(Partition.of(1), Offset.of(10))
+                .put(Partition.of(2), Offset.of(20))
+                .build());
+
+    assertThat(future.get()).isEqualTo(aggregate);
```

Comment:
You'll have to change this when you put it in beam: they won't add truth as a dependency. Use the generic mockito asserts.

Reply:
I switched to using the stuff from

import static org.junit.Assert.*;

Is that what you meant?